src/ethereum_test_tools/common/types.py

```diff
@@ -637,6 +709,83 @@ def verify_post_alloc(self, got_alloc: "Alloc"):
                 else:
                     raise Alloc.MissingAccount(address)
 
+    def deploy_contract(
+        self,
+        code: BytesConvertible,
+        *,
+        storage: Storage
+        | Dict[StorageKeyValueTypeConvertible, StorageKeyValueTypeConvertible] = {},
+        balance: NumberConvertible = 0,
+        nonce: NumberConvertible = 1,
+        address: Address | None = None,
+        label: str | None = None,
+    ) -> Address:
+        """
+        Deploy a contract to the allocation.
+
+        Warning: `address` parameter is a temporary solution to allow tests to hard-code the
+        contract address. Do NOT use in new tests as it will be removed in the future!
+        """
+        if address is not None:
+            assert self.alloc_mode == AllocMode.PERMISSIVE, "address parameter is not supported"
+            assert address not in self, f"address {address} already in allocation"
+            contract_address = address
+        else:
+            current_address = self.start_contract_address
+            while current_address in self:
+                current_address += self.contract_address_increments
+            contract_address = Address(current_address)
+
+        if self.alloc_mode == AllocMode.STRICT:
+            assert Number(nonce) >= 1, "impossible to deploy contract with nonce lower than one"
+
+        self[contract_address] = Account(
+            nonce=nonce,
+            balance=balance,
+            code=code,
+            storage=storage,
+        )
+        if label is None:
+            # Try to deduce the label from the code
+            frame = inspect.currentframe()
+            if frame is not None:
+                caller_frame = frame.f_back
+                if caller_frame is not None:
+                    code_context = inspect.getframeinfo(caller_frame).code_context
+                    if code_context is not None:
+                        line = code_context[0].strip()
+                        if "=" in line:
+                            label = line.split("=")[0].strip()
+
+        contract_address.label = label
+        return contract_address
+
+    def fund_sender(self, amount: NumberConvertible, label: str | None = None) -> Sender:
+        """
+        Fund a sender with a given amount to be able to afford transactions.
+        """
+        for sender in SENDERS:
+            if sender not in self:
+                self[sender] = Account(
+                    nonce=0,
+                    balance=amount,
+                )
+                return sender.copy()
+        raise ValueError("no more senders available")
+
+    def fund_address(self, address: Address, amount: NumberConvertible):
```

Comment:
I still like the `fund_address` because it translates better in the context of live devnets.

Although if we get picky about it, in that context we are sending a funding transaction to the address, which is technically calling the account, and might even revert.